utils/templates/validation.py.jinja

```diff
@@ -0,0 +1,31 @@
+# Automatically generated file. Do not modify.
+
+# Table for connectivity
+# "Basin": ["LinearResistance"] means that the downstream of basin can be LinearResistance only
+
+node_type_connectivity: dict[str, list[str]] = {
+{% for n in nodes %}
+    '{{n[:name]}}': [
+    {% for value in n[:connectivity] %}
+        '{{ value }}',
+    {% end %}],
```

Comment:
https://mommawatasu.github.io/OteraEngine.jl/dev/tutorial/#White-Space-Control

Reply:
Good source! In this case the white space trimming doesn't do much, because after ruff formatting the code, the code looks the same.